packages/utilities/src/ProcessUtils.ts

```diff
@@ -142,4 +143,27 @@ export class ProcessUtils {
             await require("child_process").spawn(editor, [filePath], { stdio: "inherit" });
         }
     }
+
+    /**
+     * Spawn a process with arguments and throw an error if the process fails.
+     * Parameters are same as `child_process.spawnSync` (see Node.js docs).
+     * Use this method if you want the safe argument parsing of `spawnSync`
+     * combined with the smart output handling of `execSync`.
+     * @returns Contents of stdout as buffer or string
+     */
+    public static execAndCheckOutput(command: string, args?: string[], options?: SpawnSyncOptions): Buffer | string {
+        // Implementation based on the child_process module
+        // https://github.com/nodejs/node/blob/main/lib/child_process.js
+        const result = spawnSync(command, args, options);
+        if (result.error != null) {
+            throw result.error;
+        } else if (result.status !== 0) {
+            let msg = `Command failed: ${command} ${args.join(" ")}`;
+            if (result.stderr?.length > 0) {
+                msg += `\n${result.stderr.toString()}`;
+            }
+            throw new Error(msg);
```

Comment:
Should we throw an ImperativeError?

There may be other places where we throw non Imperative errors. For example:
- `throw new Error("Failed to uninstall plugin, install folder still exists:\n  " + installFolder);`

Reply:
This method is intended to be low-level and mimic the error handling behavior of `child_process.execSync`. If we throw an Imperative error that wraps the original error, it may affect how the original error gets logged. I'm open to switching if we think that using vanilla JS errors is bad practice here 🙂